### PR TITLE
Fixing the issue #1123

### PR DESCRIPTION
**Pull request by minudika**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java [hunk 1]
```diff
@@ -1294,39 +1286,28 @@ private void updateArtifactCount(String businessRuleUUID, int artifactCount)
...
-                (deploymentStatus != TemplateManagerConstants.PARTIALLY_DEPLOYED)) {
-            if (businessRule instanceof BusinessRuleFromTemplate) {
-                BusinessRuleFromTemplate businessRuleFromTemplate = (BusinessRuleFromTemplate) businessRule;
-                String instanceLimit = getRuleTemplate(businessRuleFromTemplate).getInstanceCount();
-                count = getInstanceCount(businessRuleFromTemplate.getRuleTemplateUUID());
+        if (businessRule instanceof BusinessRuleFromTemplate) {
+            BusinessRuleFromTemplate businessRuleFromTemplate = (BusinessRuleFromTemplate) businessRule;
+            String instanceLimit = getRuleTemplate(businessRuleFromTemplate).getInstanceCount();
+            count = getInstanceCount(businessRuleFromTemplate.getRuleTemplateUUID());
+            if ("one".equalsIgnoreCase(instanceLimit) && count >= 1) {
```
Comment (tishan89) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: move to constants

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java [hunk 2]
```diff
@@ -1294,39 +1286,28 @@ private void updateArtifactCount(String businessRuleUUID, int artifactCount)
...
+                throw new TemplateInstanceCountViolationException("Rule Template '" +
+                        ((BusinessRuleFromTemplate) businessRule).getRuleTemplateUUID() + "' can be instantiated " +
```
Comment (tishan89) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: use already casted variable

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java [hunk 3]
```diff
@@ -1294,39 +1286,28 @@ private void updateArtifactCount(String businessRuleUUID, int artifactCount)
...
+                        "only once and has already been instantiated.");
+            }
+        } else if (businessRule instanceof BusinessRuleFromScratch) {
+            BusinessRuleFromScratch businessRuleFromScratch = (BusinessRuleFromScratch) businessRule;
+            List<RuleTemplate> ruleTemplates =
+                    (List<RuleTemplate>) getInputOutputRuleTemplates(businessRuleFromScratch);
```
Comment (tishan89) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: Why casting here? Does this method return generics. Cant we make the return specific?

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java [hunk 4]
```diff
@@ -1294,39 +1286,28 @@ private void updateArtifactCount(String businessRuleUUID, int artifactCount)
...
+            String instanceLimit;
+            for (RuleTemplate ruleTemplate : ruleTemplates) {
+                instanceLimit = (ruleTemplate.getInstanceCount());
```
Comment (tishan89) [inline components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: remove extra bracket

Comment (minudika) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: Fixed with PR #1205

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java (hunk 3 above)
Comment (minudika) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: Fixed with PR #1205

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java (hunk 2 above)
Comment (minudika) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: Fixed with PR #1205

File: components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java (hunk 1 above)
Comment (minudika) [reply components/org.wso2.carbon.business.rules.core/src/main/java/org/wso2/carbon/business/rules/core/services/TemplateManagerService.java]: Fixed with PR #1205

**Pull request merged**